### PR TITLE
Use `when` instead of `if` in Android event serialization

### PR DESCRIPTION
**Pull request by strangesource**
## Description
Multiple `if` clause without `else if` do not prevent duplicate declaration of events and is not readable.

## Changes
Use a `when` instead of multiple `if` in order to improve readability and avoid duplicate event declaration.

The changes do not modify any functionality.

## Checklist
- [ ] 🗒 `CHANGELOG` entry

### CODE REVIEW

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt [hunk 1]
```diff
@@ -396,39 +396,54 @@ class JsonConverter {
...
+                    json.putMap("subtitleTrack", fromSubtitleTrack(event.subtitleTrack))
+                }
+
+                is SourceEvent.SubtitleTrackChanged -> {
+                    json.putMap("oldSubtitleTrack", fromSubtitleTrack(event.oldSubtitleTrack))
+                    json.putMap("newSubtitleTrack", fromSubtitleTrack(event.newSubtitleTrack))
+                }
+
+                else -> {
+                    // Event is not supported yet or does not have any additional data
```
Comment (matamegger) [inline android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt:445]: A `Log` would be nice :)

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt [hunk 2]
```diff
@@ -443,86 +458,112 @@ class JsonConverter {
...
+                    json.putDouble("timeOffset", event.timeOffset)
+                }
+
+                is PlayerEvent.VideoPlaybackQualityChanged -> {
+                    json.putMap("newVideoQuality", fromVideoQuality(event.newVideoQuality))
+                    json.putMap("oldVideoQuality", fromVideoQuality(event.oldVideoQuality))
+                }
+
+                else -> {
+                    // Event is not supported yet or does not have any additional data
```
Comment (matamegger) [inline android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt:565]: A `Log` would be nice :)

**Review by matamegger — APPROVED** [review]
Nice improvement!

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt (hunk 1 above)
Comment (strangesource) [reply android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt:445]: I don't think it's a good idea to log here as there are many events that just don't have any additional data that would suddenly fill up the logs even though they are fully supported. 
In order to have a meaningful log we would need to explicitly include those in the 'when' entry as supported, but I consider this out of scope for this quick improvement.

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt (hunk 2 above)
Comment (strangesource) [reply android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt:565]: https://github.com/bitmovin/bitmovin-player-react-native/pull/240#discussion_r1329690343

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt (hunk 1 above)
Comment (matamegger) [reply android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt:445]: Makes sense!

**Pull request merged**